apollo-router/src/metrics/aggregation.rs

```diff
@@ -214,12 +214,12 @@ impl<T: Copy> SyncCounter<T> for AggregateCounter<T> {
 }
 
 pub(crate) struct AggregateObservableCounter<T> {
-    delegates: Vec<ObservableCounter<T>>,
+    delegates: Vec<(ObservableCounter<T>, Option<DroppingUnregister>)>,
```

Comment:
Maybe a very dumb question but why is it not included directly in the `ObservableCounter` ? Like only having a single type responsible to the drop impl ?

Reply:
We can take the suggestion back to Otel rust. I agree that it seems hard to clean observables.
Also would want to suggest that noop instrument not give the cryptic error message when passed to register_callback.